compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1368,7 +1369,100 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         "the following trait bounds were not satisfied:\n{bound_list}"
                     ));
                 }
-                suggested_derive = self.suggest_derive(&mut err, unsatisfied_predicates);
+
+                let mut suggest_derive = true;
+                for (pred, _, cause) in unsatisfied_predicates {
+                    let Some(ty::PredicateKind::Clause(ty::ClauseKind::Trait(trait_pred))) =
+                        pred.kind().no_bound_vars()
+                    else {
+                        continue;
+                    };
+                    let (adt, params) = match trait_pred.self_ty().kind() {
+                        ty::Adt(adt, params) if adt.did().is_local() => (*adt, params),
+                        _ => continue,
+                    };
+                    if self
+                        .tcx
+                        .all_impls(trait_pred.def_id())
+                        .filter_map(|imp_did| {
+                            self.tcx.impl_trait_header(imp_did).map(|h| (imp_did, h))
+                        })
+                        .filter(|(did, header)| {
+                            let imp = header.trait_ref.instantiate_identity();
+                            let impl_adt = match imp.self_ty().ty_adt_def() {
+                                Some(impl_adt) if adt.did().is_local() => impl_adt,
+                                _ => return false,
+                            };
+                            header.polarity == ty::ImplPolarity::Positive
+                                && impl_adt == adt
+                                && self.tcx.is_automatically_derived(*did)
```

Comment:
I don't think this check is sufficient -- I can slap `automatically_derived` onto any impl I want without any problems. There is no guarantee that this is actually a derived impl in a way that makes this logic sufficient.

Reply:
Added a check only for built-in derives.